src/main/java/parsing/Parser.java

```diff
@@ -14,21 +16,27 @@
 import java.io.FileWriter;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.LineNumberReader;
 import java.io.OutputStreamWriter;
 
 /**
  * Created by 101010.
  */
-public final class Parser {
+public class Parser {
     /**
      * Initial Parser.
      */
     private static Parser instance = null;
 
     /**
+     * Thread the parser is running in.
+     */
+    private static Thread parser;
+
+    /**
      * Constructor of the parser.
      */
-    private Parser() { }
+    private Parser() {}
```

Comment:
My inspection does not give the warnings you mention, which is odd since they should be there.